### PR TITLE
Implement news draw function

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/src/Windows/BuildVehicle.cpp [hunk 1]
```diff
@@ -1446,7 +1445,7 @@ namespace OpenLoco::Ui::Windows::BuildVehicle
     }
 
     // 0x4B7741
-    static void drawVehicleOverview(Gfx::RenderTarget* rt, int16_t vehicleTypeIdx, CompanyId company, uint8_t eax, uint8_t esi, Ui::Point offset)
+    void drawVehicleOverview(Gfx::RenderTarget* rt, int16_t vehicleTypeIdx, CompanyId company, uint8_t eax, uint8_t esi, Ui::Point offset)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/BuildVehicle.cpp:1448]: Not a fan of having to exposing this for use out of the window, but then again this is one massive function to implement. It'll have to do for now.

**Review by AaronVanGeffen — COMMENTED** [review]
Any ideas for how best to trigger a news message that uses this routine?

**Comment by ZehMatt** [comment]
> Any ideas for how best to trigger a news message that uses this routine?

Vehicle crash will do

**Review by ZehMatt — APPROVED** [review]

**Comment by AaronVanGeffen** [comment]
> Vehicle crash will do

I don't think it does -- that doesn't draw the company owner/competitor object, which the code refers to.

**Comment by ZehMatt** [comment]
> > Vehicle crash will do
> 
> I don't think it does -- that doesn't draw the company owner/competitor object, which the code refers to.

I placed a breakpoint on the function and it hits when the news element shows up.

**Comment by AaronVanGeffen** [comment]
> I placed a breakpoint on the function and it hits when the news element shows up.

I'm guessing you watched a news message with `unk == -3` whereas the case I'm looking at is `unk == -2`.

File: src/OpenLoco/src/Windows/News/News.cpp [hunk 2]
```diff
@@ -458,11 +459,40 @@ namespace OpenLoco::Ui::Windows::NewsWindow
...
+                const int32_t unk = i == 0 ? *_dword_525CD0 : *_dword_525CD8;
+                if (unk == -2 && itemSubject != 0xFFFFU)
+                {
+                    const auto* company = CompanyManager::get(CompanyId(itemSubject));
+                    const auto* competitorObj = ObjectManager::get<CompetitorObject>(company->competitorId);
+                    const auto imageIndexBase = competitorObj->images[company->ownerEmotion];
+
+                    const ImageId imageId(imageIndexBase + 1, company->mainColours.primary);
+                    const auto x = self->x + viewWidget.width() - 32;
+                    const auto y = self->y + viewWidget.height() - 32;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/News/News.cpp:476]:
```suggestion
                    const auto x = self->x + viewWidget.midX() - 31;
                    const auto y = self->y + viewWidget.midY() - 31;
```

File: src/OpenLoco/src/Windows/News/News.cpp [hunk 3]
```diff
@@ -458,11 +459,40 @@ namespace OpenLoco::Ui::Windows::NewsWindow
         // 0x0042A136
         static void sub_42A136(Window* self, Gfx::RenderTarget* rt, Message* news)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/News/News.cpp:460]: Perhaps we can rename this to drawNewsSubjects?

File: src/OpenLoco/src/Windows/BuildVehicle.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Windows/BuildVehicle.cpp:1448]: Yeah it needs to be moved to somewhere else. Just not sure where.

File: src/OpenLoco/src/Windows/News/News.cpp (hunk 3 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Windows/News/News.cpp:460]: hmm its more like drawNonViewportNewsSubjects

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Windows/News/News.cpp:460]: Hmm, that's a bit long. I suppose `drawOtherNewsSubjects` is too vague?

**Review by AaronVanGeffen — APPROVED** [review]

File: src/OpenLoco/src/Windows/News/News.cpp (hunk 3 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Windows/News/News.cpp:460]: drawNewsSubjectImages ? would that be better?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Windows/News/News.cpp:460]: Yeah, sounds good to me!

**Pull request merged**